### PR TITLE
Task/jenkins 42108 filter list component

### PR DESCRIPTION
**Pull request by cliffmeyers**
**Description**
- See [JENKINS-42108](https://issues.jenkins-ci.org/browse/JENKINS-42108).
- Incorporates changes from #148 for the icon text input which can be ignored in this PR. Once #148 is approved let me know, and I'll remerge master to get this one cleaned up.

**Submitter checklist**
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [ ] Appropriate unit or acceptance tests or explanation to why this change has no tests

**Reviewer checklist**
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

@reviewbybees

### CODE REVIEW

**Comment by michaelneale** [comment]
:bee: from me but @sophistifunk @kzantow ?

File: src/js/components/FilterableList.jsx [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+                    {this.props.children}
+                </List>
+                }
+            </div>
+        );
+    }
+}
+
+FilterableList.propTypes = {
+    // from List
```
Comment (sophistifunk) [inline src/js/components/FilterableList.jsx]: If this isn't a subset, could we use `...List.propTypes` so they automatically stay in sync?

**Comment by sophistifunk** [comment]
LGTM

yatta

File: src/js/components/FilterableList.jsx (hunk 1 above)
Comment (cliffmeyers) [inline src/js/components/FilterableList.jsx]: Adding a TODO to address that this week. At this point I just need to get this sucker merged so I can move forward with some testing. Thanks, will definitely make sure this gets sorted :)

**Pull request merged**